File: Donggyo/week5/day21/연속된부분수열의합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = {};
        int N = sequence.length;
        int sum = 0;
        int left = 0;
        int min_length = Integer.MAX_VALUE;
        
        for(int right = 0; right< N; right++){
            sum += sequence[right];
            
            while(sum > k){
                sum -= sequence[left++];
            }
            if(sum == k && right-left < min_length){
                answer = new int[]{left, right};
                min_length = right-left;
            }
        }
        return answer;
    }
}

File: Jiho/week5/day21/연속된 부분 수열의 합.js
function solution(sequence, k) {
  let start_index = 0;
  let end_index = 0;
  let sum = sequence[0];
  let answer = null;

  while (end_index < sequence.length) {
    if (sum === k) {
      if (!answer || end_index - start_index < answer[1] - answer[0]) {
        answer = [start_index, end_index];
      }
    }

    if (sum < k) {
      end_index++;
      if (end_index < sequence.length) {
        sum += sequence[end_index];
      }
    } else {
      sum -= sequence[start_index];
      start_index++;
    }
  }

  return answer;
}
